tasks/20240223-composable-stable-pool-v6/test/test.fork.ts

```diff
@@ -0,0 +1,525 @@
+      expect(await pool.inRecoveryMode()).to.be.true;
+    });
+
+    it('can exit via recovery mode', async () => {
+      const bptBalance = await pool.balanceOf(owner.address);
+      expect(bptBalance).to.gt(0);
+
+      const vaultUSDCBalanceBeforeExit = await usdc.balanceOf(vault.address);
+      const ownerUSDCBalanceBeforeExit = await usdc.balanceOf(owner.address);
+
+      const { tokens: registeredTokens } = await vault.getPoolTokens(poolId);
+
+      const userData = BasePoolEncoder.recoveryModeExit(bptBalance);
+      await vault.connect(owner).exitPool(poolId, owner.address, owner.address, {
+        assets: registeredTokens,
+        minAmountsOut: Array(registeredTokens.length).fill(0),
+        fromInternalBalance: false,
+        userData,
+      });
+
+      const remainingBalance = await pool.balanceOf(owner.address);
+      expect(remainingBalance).to.equal(0);
+
+      const vaultUSDCBalanceAfterExit = await usdc.balanceOf(vault.address);
+      const ownerUSDCBalanceAfterExit = await usdc.balanceOf(owner.address);
+
+      expect(vaultUSDCBalanceAfterExit).to.lt(vaultUSDCBalanceBeforeExit);
+      expect(ownerUSDCBalanceAfterExit).to.gt(ownerUSDCBalanceBeforeExit);
+    });
+  });
+
+  describe('create2 functionality', () => {
+    it('can be deployed with a different salt', async () => {
+      const pool = await createPool(tokens, ZERO_ADDRESS, false, ZERO_BYTES32);
+      const pool2 = await createPool(tokens, ZERO_ADDRESS, false, ONES_BYTES32);
+
+      expect(pool2.address).to.not.equal(pool.address);
+    });
+  });
+
+  it('cannot execute the contract halves', async () => {
+    const { contractA, contractB } = await factory.getCreationCodeContracts();
+
+    const txA = {
+      to: contractA,
+      value: ethers.utils.parseEther('0.001'),
+    };
+
+    const txB = {
+      to: contractB,
+      value: ethers.utils.parseEther('0.001'),
+    };
+
+    await expect(owner.sendTransaction(txA)).to.be.reverted;
+    await expect(owner.sendTransaction(txB)).to.be.reverted;
+  });
+
+  describe('pause window', () => {
+    const EXPECTED_PAUSE_WINDOW = 4 * 365 * DAY;
+    const EXPECTED_BUFFER_PERIOD = 180 * DAY;
+
+    sharedBeforeEach(async () => {
+      // Reset timestamp and double check that we're close to factory deployment time.
+      expect(await currentTimestamp()).to.be.lt(factoryDeploymentTimestamp.add(MINUTE));
+    });
+
+    it('can be paused 4 years after factory deployment', async () => {
+      const pool = await createPool(tokens, ZERO_ADDRESS, false, ZERO_BYTES32);
+      await authorizer.connect(govMultisig).grantRole(await actionId(pool, 'pause'), govMultisig.address);
+      await advanceTime(EXPECTED_PAUSE_WINDOW - MINUTE);
+      const tx = await pool.connect(govMultisig).pause();
+      expectEvent.inReceipt(await tx.wait(), 'PausedStateChanged', { paused: true });
+    });
+
+    it('cannot be paused 4 years after factory deployment', async () => {
```

Comment:
```suggestion
    it('cannot be paused more than 4 years after factory deployment', async () => {
```
The two test names sounded inconsistent.